File: src/main/java/dev/psygamer/wireframe/util/collection/FreezableLinkedHashSet.java
package dev.psygamer.wireframe.util.collection;

import java.util.Collection;
import java.util.LinkedHashSet;

public class FreezableLinkedHashSet <E> extends FreezableSet<E> {
	
	public FreezableLinkedHashSet() {
		super(new LinkedHashSet<>());
	}
	
	public FreezableLinkedHashSet(final int initialCapacity) {
		super(new LinkedHashSet<>(initialCapacity));
	}
	
	public FreezableLinkedHashSet(final int initialCapacity, final float loadFactor) {
		super(new LinkedHashSet<>(initialCapacity, loadFactor));
	}
	
	public FreezableLinkedHashSet(final Collection<? extends E> set) {
		super(new LinkedHashSet<>(set));
	}
	
	@Override
	public FreezableSet<E> copy() {
		final FreezableSet<E> set = new FreezableLinkedHashSet<>(this);
		
		if (isFrozen())
			set.freeze();
		
		return set;
	}
}

File: src/main/kotlin/dev/psygamer/wireframe/util/collection/FreezableLinkedHashSet.kt
package dev.psygamer.wireframe.util.collection

class FreezableLinkedHashSet<E> : FreezableSet<E> {
	
	constructor() : super(LinkedHashSet<E>())
	constructor(initialCapacity: Int) : super(LinkedHashSet<E>(initialCapacity))
	constructor(initialCapacity: Int, loadFactor: Float) : super(LinkedHashSet<E>(initialCapacity, loadFactor))
	constructor(set: Set<E>) : super(LinkedHashSet<E>(set))
}